### PR TITLE
AGMS-990: Authenticate showcase

### PR DESCRIPTION
**Pull request by b1zzu**
**Close**:
* https://issues.jboss.org/browse/AGMS-990

**Changes**:
* Migrate from protractor to webdriverio. Why:
   * Protractor works fine in browser with plain angular, but for login we have to switch window to a non Angular app, and this is quite tricky in protractor.
   * We are already using webdriverio for the device tests and we could reuse the same login function for both
   * We can reuse knowledge and helpers between device and showcase test set
   * Protractor prefer Jasmin and is not directly supporting  mocha&chai 
* Login before start testing
* Verify that the profile page

### CODE REVIEW

File: showcase/util/init.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import { login } from "./ login";
```
Comment (jhellar) [inline showcase/util/init.ts]: Is the space here typo?

File: showcase/scripts/prepare.sh [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+# find mdc namespace
+if oc get mobileclients -n openshift-mobile-developer-console &>/dev/null; then
+  MDC_NAMESPACE=openshift-mobile-developer-console
+else
+  MDC_NAMESPACE=mobile-developer-console
+fi
+
+# cleanup
+oc delete mobileclient $APP_NAME -n $MDC_NAMESPACE || true
+oc get projects | grep "$SYNC_NAMESPACE_PREFIX-" | awk '{print $1}' | xargs -r -L1 oc delete project
```
Comment (jhellar) [inline showcase/scripts/prepare.sh]: `-r` for xargs is not supported on macs. Can this be changed/removed?

Comment (b1zzu) [reply showcase/scripts/prepare.sh]: But without it on linux will fail if there are no projects matching, maybe we shouldn't use xargs at all?

**Comment by b1zzu** [comment]
> sync should be protected with keycloak
This depends on how the showcase is deployed by the `prepare.sh` script, so I think is better to start trying it on the device test set and then bring it to the showcase, but I open a task for it to keep track

> check sync functionality
There is a task for automating the create/delete/update functionality (https://issues.jboss.org/browse/AGMS-992), but for now, I wouldn't automate the offline or subscription, create stable UI test take a lot of time, I would prefer to focus our efforts on the other tests

> logout
Idem for the logout, I would ignore it for now, and it's already covered by the device tests

wdyt?

File: showcase/scripts/prepare.sh (hunk 2 above)
Comment (jhellar) [reply showcase/scripts/prepare.sh]: could we just add `|| true` at the end of the line for now?

**Comment by jhellar** [comment]
@b1zzu ok, makes sense

File: showcase/scripts/prepare.sh (hunk 2 above)
Comment (b1zzu) [reply showcase/scripts/prepare.sh]: applied workaround

File: showcase/util/init.ts (hunk 1 above)
Comment (b1zzu) [reply showcase/util/init.ts]: Good eyes :)

**Pull request merged**